### PR TITLE
Appc 935 remove dependencies from other sdk modules

### PR DESCRIPTION
**Pull request by brunovarelaAptoide**
- Removed Retrofit and RXJava Removed.
- Appcoins_net module created
- AppcoinsClient Created
- AppcoinsHTTPClient Creted with 2 operations: PING , GET
- Log Interceptor adapted
- Wallet Utils Adapted.
- Change: only if there is a campaign the proof is sent.

### CODE REVIEW

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/ip/IpApi.java [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
-import io.reactivex.Single;
-import okhttp3.OkHttpClient;
-import retrofit2.Retrofit;
-import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
-import retrofit2.converter.jackson.JacksonConverterFactory;
-import retrofit2.http.GET;
 
 public interface IpApi {
-
+/*
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/ip/IpApi.java]: remove commented code

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/ip/IpResponse.java [hunk 2]
```diff
@@ -1,19 +1,29 @@
 package com.asf.appcoins.sdk.ads.ip;
 
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.annotation.JsonProperty;
 
-@JsonInclude(JsonInclude.Include.NON_NULL) public class IpResponse {
+public class IpResponse {
+  /*
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/ip/IpResponse.java]: remove commented code

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/campaign/CampaignMapper.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    if (found) {
+      String val = matcher.group(1);
+      //TODO should remove quotes and commas on the regex itself!
+      val = val.replaceAll("\"", "");
+      val = val.replaceAll(",", "");
+      return val;
+    }
+
+    //TODO throw error
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/campaign/CampaignMapper.java]: do the todo

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 4]
```diff
@@ -56,14 +41,14 @@
...
-  private final CampaignService campaignService;
+  private final AppcoinsClient appcoinsClient;
   /** The connector with the wallet service, receiver of the messages of the PoA. */
   private PoAServiceConnector poaConnector;
   /** The application context */
   private Context appContext;
   /** integer used to identify the network to wich we are connected */
   private int network = 0;
-  private CompositeDisposable compositeDisposable;
+  //private CompositeDisposable compositeDisposable;
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: remove commented code

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 5]
```diff
@@ -206,7 +170,7 @@ public void finishProcess() {
     final AppPreferences appPreferences = new AppPreferences(appContext);
     appPreferences.remove(PREFERENCE_WALLET_PCKG_NAME);
     poaConnector.disconnectFromService(appContext);
-    compositeDisposable.clear();
+    //compositeDisposable.clear();
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: remove commented code

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 6]
```diff
@@ -299,29 +268,18 @@ private void checkPreferencesForPackage() {
...
-  private ObservableSource<? extends Integer> retryIfNetworkAvailable(Boolean hasInternet) {
-    return hasInternet ? Observable.just(0) : Observable.empty();
-  }
-
   @Override public void onBecameForeground(Activity activity) {
     foreground = true;
 
     if (!processing) {
-      this.compositeDisposable = new CompositeDisposable();
+      //this.compositeDisposable = new CompositeDisposable();
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: remove commented code

File: appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/AppCoinsAddressProxyBuilder.java [hunk 7]
```diff
@@ -2,9 +2,9 @@
 
 import com.asf.appcoins.sdk.contractproxy.repository.RemoteRepository;
 import okhttp3.OkHttpClient;
-import retrofit2.Retrofit;
-import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
-import retrofit2.converter.jackson.JacksonConverterFactory;
+//import retrofit2.Retrofit;
```
Comment (Trinkes) [inline appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/AppCoinsAddressProxyBuilder.java]: remove commented code

File: appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/AppCoinsAddressProxyBuilder.java [hunk 8]
```diff
@@ -25,11 +25,14 @@ public AppCoinsAddressProxySdk createAddressProxySdk() {
         baseHost = BuildConfig.MAIN_NETWORK_BACKEND_BASE_HOST;
         break;
     }
-    return new Retrofit.Builder().baseUrl(baseHost)
+    //TODO REMOVER RETROFIT
```
Comment (Trinkes) [inline appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/AppCoinsAddressProxyBuilder.java]: do the todo

File: appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/repository/RemoteRepository.java [hunk 9]
```diff
@@ -1,8 +1,8 @@
 package com.asf.appcoins.sdk.contractproxy.repository;
 
 import io.reactivex.Single;
-import retrofit2.http.GET;
-import retrofit2.http.Query;
+//import retrofit2.http.GET;
```
Comment (Trinkes) [inline appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/repository/RemoteRepository.java]: remove commented code

File: appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/repository/RemoteRepository.java [hunk 10]
```diff
@@ -12,9 +12,10 @@ public RemoteRepository(ApiProvider apiProvider) {
   }
 
   public Single<String> getAddress(Contracts contract, int chainId) {
-    return apiProvider.getApi(chainId)
-        .getAddress(contract.value)
-        .map(ProxyResponse::getWallet);
+    return null;
```
Comment (Trinkes) [inline appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/repository/RemoteRepository.java]: never return null on Rxjava

File: appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/repository/RemoteRepository.java [hunk 11]
```diff
@@ -27,7 +28,8 @@ public RemoteRepository(ApiProvider apiProvider) {
   }
 
   public interface Api {
-    @GET("appc/proxyaddress") Single<ProxyResponse> getAddress(@Query("contract") String contract);
+    //TODO REMOVER RETROFIT
```
Comment (Trinkes) [inline appcoins-contract-proxy/src/main/java/com/asf/appcoins/sdk/contractproxy/repository/RemoteRepository.java]: do the todo

File: gradle.properties [hunk 12]
```diff
@@ -9,4 +9,24 @@ ROPSTEN_NETWORK_PROXY_CONTRACT_ADDRESS=0x3CA30A86d04e65E6E388922deCe3eBD0F100F5d
...
+BDS_WALLET_PACKAGE_NAME_DEV="com.appcoins.wallet.dev"
+GOOGLE_IAB_BIND_ACTION=com.android.vending.billing.InAppBillingService.BIND
+GOOGLE_IAB_BIND_PACKAGE=com.android.vending
+GOOGLE_IAB_PERMISSION=com.android.vending.BILLING
+GOOGLE_IAB_KEY=CONSTRUCT_YOUR_GOOGLE_KEY_AND_PLACE_IT_HERE
+GOOGLE_IAB_UPDATE_ACTION=com.android.vending.billing.PURCHASES_UPDATED
+APPCOINS_IAB_BIND_ACTION=com.appcoins.wallet.iab.action.BIND
+APPCOINS_IAB_BIND_PACKAGE=com.appcoins.wallet
+APPCOINS_IAB_PERMISSION=com.appcoins.BILLING
+APPCOINS_IAB_KEY=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAw+Qen4gtCi1triCxdb2svOsZ7OKJu19cjVDO2MNuvSw69Q+RQGYUYXQLfoFM6Nzk7tlbhIKjN2YC7/ZoxDFTzoHh1WNbLIn1X+D5p33fAUgd6vzrJLG0eLQ4A5MmXCAnaalCpgeA3BoRfmUr+ED09ycRXEhS/IU12bTN0pO3trm28SOGAkwBCsGYPBBlPvpRiydhvrcIbpfB3oEMrjvlpXo+9k8mSO0qk03gnrGp5N6inQe6k8qDQ+sdFQYPsdiDo//EGyzLwoLBEzg31DSObDJprz7rgHR2z1lDPkHf7NALD8/a49cS21SPX3IPY+YX9NI8HEALh/eiRjhQmqO5wwIDAQAB
```
Comment (Trinkes) [inline gradle.properties]: remove keys from version control

File: gradle/versions.gradle [hunk 13]
```diff
@@ -41,6 +41,7 @@ ver.spongycastle_prov = "1.58.0.0"
 ver.jackson = "2.9.5"
 
 // https://mvnrepository.com/artifact/com.squareup.retrofit2/retrofit
+//TODO REMOVER RETROFIT
```
Comment (Trinkes) [inline gradle/versions.gradle]: do the todo

File: toolbox/build.gradle [hunk 14]
```diff
@@ -1,42 +1,126 @@
...
 
-  defaultConfig {
-    applicationId "com.appcoins.toolbox"
+  flavorDimensions "iab", "network"
 
-    minSdkVersion MIN_SDK_VERSION
-    targetSdkVersion TARGET_SDK_VERSION
+  defaultConfig {
+    applicationId "com.appcoins.trivialdrivesample"
+    minSdkVersion 26
```
Comment (Trinkes) [inline toolbox/build.gradle]: use the project's values(min sdk 21)

File: toolbox/src/main/AndroidManifest.xml [hunk 15]
```diff
@@ -1,8 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.asf.appcoins.toolbox">
-
+  <uses-permission android:name="android.permission.INTERNET" />
   <application
+      android:usesCleartextTraffic="true"
```
Comment (Trinkes) [inline toolbox/src/main/AndroidManifest.xml]: why would you want this?

File: toolbox/src/main/java/com/asf/appcoins/toolbox/App.java [hunk 16]
```diff
@@ -2,16 +2,25 @@
 
 import android.app.Application;
 import android.content.pm.PackageManager;
+import android.os.Build;
+import android.util.Log;
+import com.asf.appcoins.sdk.ads.AppCoinsAds;
 
 public class App extends Application {
 
+  private final String developerAddress = BuildConfig.IAB_WALLET_ADDR;
```
Comment (Trinkes) [inline toolbox/src/main/java/com/asf/appcoins/toolbox/App.java]: not used

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsClient.java [hunk 17]
```diff
@@ -0,0 +1,49 @@
+package com.appcoins.net;
+
+public class AppcoinsClient implements AppcoinsConnection {
+
+  private final String packageName;
+  private final int versionCode;
+  private final Interceptor interceptor;
+  private String serviceUrl;
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsClient.java]: why isn't this final?

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+        getResponseHandler.getResponseHandler(response);
+        break;
+      case PING_CONNECTION:
+        boolean result = Ping();
+        getResponseHandler.getResponseHandler(result);
+        break;
+    }
+  }
+
+  @Override public String Get() {
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java]: use a meaningful name on methods

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+  public AppcoinsHTTPClient(String serviceUrl, Interceptor interceptor, String params,
+      int operation, GetResponseHandler getResponseHandler) {
+    this.serviceUrl = serviceUrl;
+    this.params = params;
+    this.getResponseHandler = getResponseHandler;
+    this.interceptor = interceptor;
+    this.operation = operation;
+  }
+
+  @Override public void run() {
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java]: why don't you implement this in two different classes?

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java [hunk 20]
```diff
@@ -0,0 +1,119 @@
...
+        return response;
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return response;
+  }
+
+  @Override public boolean Ping() {
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java]: methods should not start with capital letter
give a better name to this method, when you call ping, you don't expect to receive a boolean

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java [hunk 21]
```diff
@@ -0,0 +1,119 @@
...
+    try {
+      url = new URL("http://www.google.com");
+      urlConnection = (HttpURLConnection) url.openConnection();
+      int responseCode = urlConnection.getResponseCode();
+      if (responseCode == HttpURLConnection.HTTP_OK) {
+        result = true;
+      }
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+    } catch (IOException e) {
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java]: if there is an error, how do you handle it if your're only logging it here?

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsConnectionQuerys.java [hunk 22]
```diff
@@ -0,0 +1,8 @@
+package com.appcoins.net;
+
+import java.io.IOException;
+
+interface AppcoinsConnectionQuerys {
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsConnectionQuerys.java]: why do you need this interface?

File: appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+    return pingGoogle();
+  }
+
+  private boolean pingGoogle() {
+
+    boolean result = false;
+    URL url = null;
+    HttpURLConnection urlConnection = null;
+    try {
+      url = new URL("http://www.google.com");
```
Comment (Trinkes) [inline appcoins_net/src/main/java/com/appcoins/net/AppcoinsHTTPClient.java]: should we ping our servers instead?

**Review by Trinkes — CHANGES_REQUESTED** [review]

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import com.asf.appcoins.sdk.ads.BuildConfig;
+import com.asf.appcoins.sdk.ads.R;
+
+public class WalletUtils {
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java]: there is other class with the same name that's not used, delete it

**Review by Trinkes — CHANGES_REQUESTED** [review]

File: microraidenj-bds/build.gradle [hunk 25]
```diff
@@ -36,7 +36,7 @@ dependencies {
   testImplementation lib.junit
   testImplementation lib.mockito_core
   androidTestImplementation lib.android_test_runner
-
+  //TODO REMOVER RETROFIT
```
Comment (Trinkes) [inline microraidenj-bds/build.gradle]: todo

File: microraidenj-bds/src/main/java/com/bds/microraidenj/ws/BDSMicroRaidenApi.java [hunk 26]
```diff
@@ -31,7 +31,7 @@ static BDSMicroRaidenApi create(boolean debug) {
     }
 
     OkHttpClient client = builder.build();
-
+    //TODO REMOVER RETROFIT
```
Comment (Trinkes) [inline microraidenj-bds/src/main/java/com/bds/microraidenj/ws/BDSMicroRaidenApi.java]: todo

File: toolbox/build.gradle [hunk 27]
```diff
@@ -1,42 +1,126 @@
...
+      keyPassword = project.APPCOINS_WALLET_KEY_PASSWORD
+    }
   }
 
   buildTypes {
-    release {
+    debug {
       minifyEnabled false
       proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+      //      applicationIdSuffix ".debug"
```
Comment (Trinkes) [inline toolbox/build.gradle]: remove commented code

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/campaign/CampaignMapper.java [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+
+    String patternStr = "(?:\"" + paramName + "\"" + "[\\s]*:[\\s]*)(.*)";
+
+    Pattern pattern = Pattern.compile(patternStr);
+    Matcher matcher = pattern.matcher(response);
+    boolean found = matcher.find();
+
+    if (found) {
+      String val = matcher.group(1);
+      //TODO should remove quotes and commas on the regex itself!
```
Comment (Trinkes) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/campaign/CampaignMapper.java]: todo

**Review by Trinkes — CHANGES_REQUESTED** [review]

File: appcoins-ads/build.gradle [hunk 29]
```diff
@@ -46,12 +52,8 @@ dependencies {
   testImplementation lib.mockito_core
 
   androidTestImplementation lib.android_test_runner
-
-  api lib.retrofit2
   api lib.tray
-  api lib.reactivenetwork_rx2
-
-  api project(path: ':appcoins-contract-proxy')
+  api project(path: ':appcoins_net')
```
Comment (JoaoRaimundo) [inline appcoins-ads/build.gradle]: Can we improve this name, since the module is generic, does it needs to be appcoins related? And "net" part, can it be more explicit.

File: appcoins_net/src/main/java/com/appcoins/net/AppCoinsClient.java [hunk 30]
```diff
@@ -0,0 +1,47 @@
+package com.appcoins.net;
+
+public class AppCoinsClient implements AppCoinsConnection {
```
Comment (JoaoRaimundo) [inline appcoins_net/src/main/java/com/appcoins/net/AppCoinsClient.java]: If this class has such relation with the ads model, does it make sense to in a different module? Its purpose doesn't exist out of the ads context.

File: appcoins_net/src/main/java/com/appcoins/net/AppCoinsConnection.java [hunk 31]
```diff
@@ -0,0 +1,8 @@
+package com.appcoins.net;
+
+interface AppCoinsConnection {
```
Comment (JoaoRaimundo) [inline appcoins_net/src/main/java/com/appcoins/net/AppCoinsConnection.java]: If this class has such relation with the ads model, does it make sense to in a different module? Its purpose doesn't exist out of the ads context.

File: appcoins_net/src/main/java/com/appcoins/net/AppCoinsHTTPClient.java [hunk 32]
```diff
@@ -0,0 +1,78 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class AppCoinsHTTPClient implements Runnable {
```
Comment (JoaoRaimundo) [inline appcoins_net/src/main/java/com/appcoins/net/AppCoinsHTTPClient.java]: Is this HttpClient AppCoins related to have its name dependent of it?

File: appcoins_net/src/main/java/com/appcoins/net/AppCoinsPingClient.java [hunk 33]
```diff
@@ -0,0 +1,50 @@
+package com.appcoins.net;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class AppCoinsPingClient extends AppCoinsHTTPClient implements Runnable {
```
Comment (JoaoRaimundo) [inline appcoins_net/src/main/java/com/appcoins/net/AppCoinsPingClient.java]: Is this PingClient AppCoins related to have its name dependent of it?

File: appcoins_net/src/main/java/com/appcoins/net/GetCampaignOperation.java [hunk 34]
```diff
@@ -0,0 +1,49 @@
...
+  public static final String PACKAGE_NAME = "packageName";
+  public static final String VER_CODE = "vercode";
+  public static final String COUNTRY_CODE = "countryCode";
+  public static final String SORT = "sort";
+  public static final String BY = "by";
+  public static final String VALID = "valid";
+  public static final String TYPE = "type";
+
+  @Override
+  public String mapParams(String packageName, String versionCode, QueryParams queryParams) {
```
Comment (JoaoRaimundo) [inline appcoins_net/src/main/java/com/appcoins/net/GetCampaignOperation.java]: Can't we use the URI (java.net.URI) to do this mapping for us?

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 35]
```diff
@@ -264,21 +226,27 @@ private void postponeSendProof() {
...
+          Log.d("Message:", "Connectivity Available");
+
+          QueryParams queryParams = new QueryParams("desc", "price", "true", "BDS");
+
+          appcoinsClient.getCampaign(queryParams, appcoinsClientResponse -> {
+            Campaign campaign = CampaignMapper.mapCampaign(appcoinsClientResponse);
+            processCampaign(campaign);
+          });
+        } else {
+          Log.d("Message:", "No Connectivity Available: Delayed.");
```
Comment (JoaoRaimundo) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: Shoul we have an only for the purpose of logging something?

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 36]
```diff
@@ -264,21 +226,27 @@ private void postponeSendProof() {
...
-        .doOnSuccess(this::processCampaign)
-        .subscribe());
+
+    if (campaignId == null) {
+      appcoinsClient.checkConnectivity(appcoinsClientResponsePing -> {
+
+        AppCoinsClientResponsePing pingResponse =
+            (AppCoinsClientResponsePing) appcoinsClientResponsePing;
+
+        if (pingResponse.HasConnection()) {
```
Comment (JoaoRaimundo) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: What happens if we don't have network? Does the ping mechanism makes any retries? Or the process is stopped until the application is closed and opened again?

File: toolbox/src/main/AndroidManifest.xml [hunk 37]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.asf.appcoins.toolbox">
-
+  <uses-permission android:name="android.permission.INTERNET" />
```
Comment (JoaoRaimundo) [inline toolbox/src/main/AndroidManifest.xml]: Do we need this permission here when it's set up already in the modules imported that have the permission set up?

File: appcoins-ads/build.gradle [hunk 38]
```diff
@@ -25,19 +27,25 @@ android {
...
   }
 
   lintOptions {
     warning 'InvalidPackage'
   }
 
   compileOptions {
-    targetCompatibility TARGET_COMPATIBILITY
-    sourceCompatibility SOURCE_COMPATIBILITY
+    targetCompatibility TARGET_COMPATIBILITY_ADS
```
Comment (JoaoRaimundo) [inline appcoins-ads/build.gradle]: Since we are no longer using lambdas, this can be removed

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/clients/GetCampaignHTTPClient.java [hunk 39]
```diff
@@ -0,0 +1,106 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class GetCampaignHTTPClient implements Runnable {
```
Comment (JoaoRaimundo) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/clients/GetCampaignHTTPClient.java]: Is this class an HTTPClient?

File: appcoins-net/src/main/AndroidManifest.xml [hunk 40]
```diff
@@ -0,0 +1,6 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (JoaoRaimundo) [inline appcoins-net/src/main/AndroidManifest.xml]: Is this file needed?

File: gradle/android.gradle [hunk 41]
```diff
@@ -5,5 +5,8 @@ ext {
 
   TARGET_COMPATIBILITY = 1.8
   SOURCE_COMPATIBILITY = 1.8
+
+  TARGET_COMPATIBILITY_ADS = 1.7
+  SOURCE_COMPATIBILITY_ADS = 1.7
```
Comment (JoaoRaimundo) [inline gradle/android.gradle]: Is this needed?

File: toolbox/build.gradle [hunk 42]
```diff
@@ -1,42 +1,126 @@
...
+  buildToolsVersion '28.0.3'
 
-  defaultConfig {
-    applicationId "com.appcoins.toolbox"
+  flavorDimensions "iab", "network"
 
-    minSdkVersion MIN_SDK_VERSION
-    targetSdkVersion TARGET_SDK_VERSION
+  defaultConfig {
+    applicationId "com.appcoins.trivialdrivesample"
```
Comment (JoaoRaimundo) [inline toolbox/build.gradle]: Please revert this to the original package name

File: toolbox/build.gradle [hunk 43]
```diff
@@ -1,42 +1,126 @@
...
+    minSdkVersion 21
+    targetSdkVersion 28
```
Comment (JoaoRaimundo) [inline toolbox/build.gradle]: Please use the variables as it was defied before.

File: toolbox/build.gradle [hunk 44]
```diff
@@ -1,42 +1,126 @@
...
+    googleiab {
+      dimension "iab"
+      buildConfigField "String", "IAB_BIND_ACTION",
+          "\"" + project.GOOGLE_IAB_BIND_ACTION + "\""
+      buildConfigField "String", "IAB_KEY",
+          "\"" + project.GOOGLE_IAB_KEY + "\""
+      buildConfigField "String", "IAB_BIND_PACKAGE",
+          "\"" + project.GOOGLE_IAB_BIND_PACKAGE + "\""
+      buildConfigField "String", "IAB_UPDATE_ACTION",
+          "\"" + project.GOOGLE_IAB_UPDATE_ACTION+ "\""
```
Comment (JoaoRaimundo) [inline toolbox/build.gradle]: Is this used on this demo app?

File: toolbox/build.gradle [hunk 45]
```diff
@@ -1,42 +1,126 @@
...
 dependencies {
-  implementation lib.appcompat_v7
-  implementation lib.constraint_layout
-
-  testImplementation lib.junit
-
-  api project(':appcoins')
+  api 'com.android.support:appcompat-v7:28.0.0'
+  api 'com.google.code.gson:gson:2.8.2'
+  api "com.google.code.gson:gson:2.8.2"
```
Comment (JoaoRaimundo) [inline toolbox/build.gradle]: is this used in here?

File: toolbox/src/main/java/com/asf/appcoins/toolbox/App.java [hunk 46]
```diff
@@ -2,16 +2,23 @@
...
   @Override public void onCreate() {
     super.onCreate();
 
     AdvertisementSdkSingleton.create(this, true);
+
     try {
-      AdvertisementSdkSingleton.getAdsSdk()
-          .init(this);
+
+      adsSdk = AdvertisementSdkSingleton.getAdsSdk();
```
Comment (JoaoRaimundo) [inline toolbox/src/main/java/com/asf/appcoins/toolbox/App.java]: Is this change needed?

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/clients/GetCampaignHTTPClient.java [hunk 47]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    return response;
+  }
+
+  private String buildURL(){
+    Uri campaignUri = Uri.parse(serviceUrl+"/campaign/listall?")
+        .buildUpon()
+        .appendQueryParameter(PACKAGE_NAME, packageName)
+        .appendQueryParameter(VER_CODE, versionCode + "")
```
Comment (JoaoRaimundo) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/clients/GetCampaignHTTPClient.java]: Why don't we use String.valueOf() instead of concatenation?

File: appcoins-ads/build.gradle (hunk 38 above)
Comment (brunovarelaAptoide) [reply appcoins-ads/build.gradle]: Removed

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/clients/GetCampaignHTTPClient.java (hunk 39 above)
Comment (brunovarelaAptoide) [reply appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/clients/GetCampaignHTTPClient.java]: Changed for GetCampaignClient.

File: appcoins-net/src/main/AndroidManifest.xml (hunk 40 above)
Comment (brunovarelaAptoide) [reply appcoins-net/src/main/AndroidManifest.xml]: Removed

File: gradle/android.gradle (hunk 41 above)
Comment (brunovarelaAptoide) [reply gradle/android.gradle]: Removed

File: appcoins_net/src/main/java/com/appcoins/net/GetCampaignOperation.java [hunk 48]
```diff
@@ -0,0 +1,49 @@
...
+    return "/campaign/listall?"
+        + PACKAGE_NAME
```
Comment (JoaoRaimundo) [inline appcoins_net/src/main/java/com/appcoins/net/GetCampaignOperation.java]: Can't we use the URI (java.net.URI) to do this mapping for us?

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 49]
```diff
@@ -48,29 +37,29 @@
...
-  private CompositeDisposable compositeDisposable;
   /** boolean indicating if we are already processing a PoA */
   private boolean processing;
-  /** The handle to keep the runnable tasks that we be running within a certain period */
+  /** The handleRetryConnection to keep the runnable tasks that we be running within a certain period */
   private Handler handler = new Handler();
   /** The runnnable taks that will be trigger periodically */
   private Runnable sendProof;
-  /** The handle to keep the runnable tasks that we be running within a certain period */
+  /** The handleRetryConnection to keep the runnable tasks that we be running within a certain period */
```
Comment (JoaoRaimundo) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: Find replace of string changed the javadoc. Please revert the change.

File: appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java [hunk 50]
```diff
@@ -48,29 +37,29 @@
...
   private PoAServiceConnector poaConnector;
   /** The application context */
   private Context appContext;
   /** integer used to identify the network to wich we are connected */
   private int network = 0;
-  private CompositeDisposable compositeDisposable;
   /** boolean indicating if we are already processing a PoA */
   private boolean processing;
-  /** The handle to keep the runnable tasks that we be running within a certain period */
+  /** The handleRetryConnection to keep the runnable tasks that we be running within a certain period */
```
Comment (JoaoRaimundo) [inline appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java]: Find replace of string changed the javadoc. Please revert the change.
